File: oop/abstractClass/java/Enemy.java
package com.cgtfarmer.example;

public class Enemy extends Character {

  public Enemy(String name, int hp, int strength, int defense) {
    super(name, hp, strength, defense);
  }

  @Override
  public float calculateDmg() {
    return (this.strength * 0.5f);
  }
}

File: oop/abstractClass/after/enemy.py
from character import Character

class Enemy(Character):
  def __init__(self, name, hp, strength, defense):
    super().__init__(name, hp, strength, defense)

  def setName(self, name):
    raise Exception('Enemies cannot change their name')

  def calculateDmg(self):
    return (super().getStrength() * 0.5)
